data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt

```diff
@@ -19,6 +22,21 @@ class SafeRepository(
     private val ethereumRepository: EthereumRepository
 ) {
 
+    val keyFlow = callbackFlow {
+        val listener = SharedPreferences.OnSharedPreferenceChangeListener { _, key -> offer(key) }
+        preferenceManager.prefs.registerOnSharedPreferenceChangeListener(listener)
+        awaitClose { preferenceManager.prefs.unregisterOnSharedPreferenceChangeListener(listener) }
```

Comment:
Nice :) My only concern is that this is a cold flow, and here it seems like a hot observable makes more sense. But it it works it works. I think we will find out when you add the tests to this.